frontend/apps/web/components/jobs/NosqlTable/NosqlTable.tsx

```diff
@@ -696,8 +793,23 @@ function EditCollection(props: EditCollectionProps): ReactElement {
     setIsEditingMapping(false);
   };
 
+  const currentColumn = data[index].column;
+  const currentSchemaTable = `${data[index].schema}.${data[index].table}`;
+
+  // filter rows that conflict with the key that is currently selected
+  const conflictRows = data.filter(
+    (obj) =>
+      obj.column === currentColumn &&
+      `${obj.schema}.${obj.table}` !== currentSchemaTable
+  );
+
+  // filter available collections by excluding those that conflict
+  const availableRows = collections.filter(
+    (item) => !conflictRows.some((obj) => `${obj.schema}.${obj.table}` === item)
+  );
```

Comment:
Since all of this is to just filter the collections, can we move this logic outside of this function to the caller and then just pass in the collections already filtered? That will keep this component simple and not require a leakage by passing in the entire data and index.